### PR TITLE
Added workload details and pod details view

### PR DESCRIPTION
**Pull request by dabasnidhi**
Workload details-
image
image

Pod details-
image

image

### CODE REVIEW

**Comment by madhuv-msft** [comment]
couple of general comments:
1) In the title of the service, we should show the status as per the figma file
2) we should see if we can show pipeline ID etc, talk to Deepak or me on this.
image
3) we should say 'Pods' rather 'Pods running' for service details pod section

**Comment by madhuv-msft** [comment]
Individual pod details:
1) colors of the 'pod details' table, label and value suppose to be different.
2) we show 'on the left' which pod is selected

File: src/WebUI/Common/KubeSummary.scss [hunk 1]
```diff
@@ -7,11 +7,16 @@
     word-wrap: normal;
 }
 
+#main-content {
```
Comment (madhuv-msft) [inline src/WebUI/Common/KubeSummary.scss]: this is very bad, if someone has an id of main-content then this going to affect them. we should use a css or parent css class, so this is not affected others.

File: src/WebUI/Pods/PodOverview.scss [hunk 2]
```diff
@@ -0,0 +1,11 @@
+@import "../Common/Common.scss";
+
+.pod-overview-key {
+    padding-left: 8px;
+    color: $secondary-text;
+}
+
+.kube-list-content .bolt-header.bolt-card-header {
```
Comment (madhuv-msft) [inline src/WebUI/Pods/PodOverview.scss]: this is specific to platform, if they change we have a breakage, shall we keep this in a file.

.bolt-header.bolt-card-header

File: src/WebUI/Pods/PodOverview.tsx [hunk 3]
```diff
@@ -14,47 +14,48 @@ import * as Resources from "../Resources";
...
     public render(): JSX.Element {
         const pod: V1Pod = this.props.pod;
         const columns: ITableColumn<any>[] = [
             {
                 id: "key",
                 name: "key",
-                width: new ObservableValue(200),
-                className: "s-key",
-                minWidth: 180,
+                width: new ObservableValue(150),
```
Comment (madhuv-msft) [inline src/WebUI/Pods/PodOverview.tsx]: we should take help from vidit on the column width

File: src/WebUI/Pods/PodOverview.tsx [hunk 4]
```diff
@@ -14,47 +14,48 @@ import * as Resources from "../Resources";
...
             { key: Resources.NodeText, value: pod.spec.nodeName || "" },
-            { key: Resources.ClusterIPText, value: "" }
+            { key: Resources.ClusterIPText, value: "" },
+            { key: Resources.ImageText, value: image },
+            { key: Resources.LabelsText, value: pod.metadata.labels || {} }
         ]);
 
         return (
-            <Card className="kube-list-content s-details depth-16" titleProps={{ text: Resources.SummaryText }}>
+            <Card className="kube-list-content s-details depth-16" titleProps={{ text: Resources.PodDetailsHeader }} contentProps={{ contentPadding: false }}>
```
Comment (madhuv-msft) [inline src/WebUI/Pods/PodOverview.tsx]: ideally, card should have this depth-16 or any other corner shapping. if not we should inform to platform team.

File: src/WebUI/Pods/PodsLeftPanel.scss [hunk 5]
```diff
@@ -1,5 +1,17 @@
...
+    .pod-left-panel-header {
+        padding-left: 20px;
+    }
+
+    .pod-left-panel-table-header {
+        padding-left: 20px;
+        margin-top: 30px;
+        margin-bottom: 30px;
+        font-size: 14px;
+        font-weight: 600;
```
Comment (madhuv-msft) [inline src/WebUI/Pods/PodsLeftPanel.scss]: we should see if we have a better way for this? 600

File: src/WebUI/Pods/PodsLeftPanel.tsx [hunk 6]
```diff
@@ -54,44 +52,14 @@ export class PodsLeftPanel extends BaseComponent<IPodsLeftPanelProperties> {
...
-                    showLines={false}
-                    singleClickActivation={false}
-                    itemProvider={tableItems}
-                    pageSize={tableItems.length}
-                    columns={columns}
-                />
-            </Card>
-        );
+    private _getHeader(): JSX.Element | null {
+        /*Add back button here when we have support for the same*/
```
Comment (madhuv-msft) [inline src/WebUI/Pods/PodsLeftPanel.tsx]: we are using format like
// todo ::

File: src/WebUI/Pods/PodsLeftPanel.tsx [hunk 7]
```diff
@@ -54,44 +52,14 @@ export class PodsLeftPanel extends BaseComponent<IPodsLeftPanelProperties> {
...
+        return (<h2 className="pod-left-panel-header">{this.props.parentName}</h2>);
```
Comment (madhuv-msft) [inline src/WebUI/Pods/PodsLeftPanel.tsx]: is h2 the right way to use here?

Comment (madhuv-msft) [inline src/WebUI/Pods/PodsLeftPanel.tsx]: is h2 the right way to use here?

File: src/WebUI/Pods/PodsRightPanel.tsx [hunk 8]
```diff
@@ -62,6 +63,10 @@ export class PodsRightPanel extends BaseComponent<IPodRightPanelProps, IPodsRigh
         );
     }
 
+    private _getHeader(): JSX.Element | null {
+        return (<h2 className="pod-right-panel-header">{(this.props.pod.metadata && this.props.pod.metadata.name) || ""}</h2>);
```
Comment (madhuv-msft) [inline src/WebUI/Pods/PodsRightPanel.tsx]: we should check pipeline code, whether they have used h2 or not?

File: src/WebUI/Pods/PodsTable.tsx [hunk 9]
```diff
@@ -51,62 +52,66 @@ export class PodsTable extends BaseComponent<IPodsTableProperties> {
...
+                    showLines={true}
                 />
             );
         }
 
         return null;
     }
 
-    private static _getColumns(showWorkloads: boolean): ITableColumn<V1Pod>[] {
+    private static _getColumns(showWorkloadsColumn: boolean): ITableColumn<V1Pod>[] {
```
Comment (madhuv-msft) [inline src/WebUI/Pods/PodsTable.tsx]: i am assuming you have taken care of the widths based on the figma, if not we should follow

File: src/WebUI/Services/ServiceDetails.scss [hunk 10]
```diff
@@ -7,7 +7,6 @@
 
         &.bolt-table-cell {
             padding-left: 0px;
```
Comment (madhuv-msft) [inline src/WebUI/Services/ServiceDetails.scss]: we should avoid overriding bolt css. if at all needed, we should keep all of them in one place, so in future we can take care of it, if we can not have a css class on our own.

File: src/WebUI/WorkLoads/WorkloadDetails.scss [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+    padding-left: 20px;
+}
+
+.workload-details-secondary-text {
+    padding-left: 20px;
+    color: $secondary-text;
+    font-weight: $fontWeightSemiBold;
+}
+
+.w-key {
```
Comment (madhuv-msft) [inline src/WebUI/WorkLoads/WorkloadDetails.scss]: too genric and not part of any parent key

File: src/WebUI/WorkLoads/WorkloadDetails.tsx [hunk 12]
```diff
@@ -0,0 +1,207 @@
...
+
+    public componentDidMount(): void {
+        const podList = this._podsStore.getState().podsList;
+        let pods: V1Pod[] = (podList && podList.items || []).filter(pod => {
+            return Utils.isOwnerMatched(pod.metadata, this.props.parentMetaData.uid);
+        });
+
+        this.setState({
+            pods: pods,
+            selectedPod: pods[0]
```
Comment (madhuv-msft) [inline src/WebUI/WorkLoads/WorkloadDetails.tsx]: make sure this [0] is not an issue, check for length

File: src/WebUI/WorkLoads/WorkloadDetails.tsx [hunk 13]
```diff
@@ -0,0 +1,207 @@
...
+                    <ResourceStatus statusProps={this.props.statusProps} customDescription={headerItem} statusSize={StatusSize.l} />
+                </div>
+            );
+        }
+
+        return null;
+    }
+
+    private static _getColumns(): ITableColumn<any>[] {
+        const columns: ITableColumn<any>[] = [
```
Comment (madhuv-msft) [inline src/WebUI/WorkLoads/WorkloadDetails.tsx]: we should talk to vidit for column sizes

File: src/WebUI/WorkLoads/WorkloadDetails.tsx [hunk 14]
```diff
@@ -0,0 +1,207 @@
...
+        this.setState({
+            pods: pods,
+            selectedPod: pods[0]
+        });
+    }
+
+    private _getMainHeading(): JSX.Element | null {
+        const metadata = this.props.parentMetaData;
+        if (metadata) {
+            const headerItem: React.ReactNode = (<h2 className="title-heading">{metadata.name}</h2>);
```
Comment (madhuv-msft) [inline src/WebUI/WorkLoads/WorkloadDetails.tsx]: please check pipeline code for this to be right header. we can avoid h2

File: src/WebUI/WorkLoads/WorkloadDetails.tsx [hunk 15]
```diff
@@ -0,0 +1,207 @@
...
+            }
+        ];
+        return columns;
+    }
+
+    private _getWorkloadDetails(): JSX.Element | null {
+        const metadata = this.props.parentMetaData;
+        if (metadata) {
+            const pipeline = Utils.getPipelineText(metadata.annotations);
+            const tableItems = new ArrayItemProvider<any>([{}, { podTemplate: this.props.podTemplate, parentMetaData: metadata }]);
```
Comment (madhuv-msft) [inline src/WebUI/WorkLoads/WorkloadDetails.tsx]: if this has more than one image we should show different text
podTemplate

also we should make this as link

File: src/WebUI/WorkLoads/WorkloadDetails.tsx [hunk 16]
```diff
@@ -0,0 +1,207 @@
...
+    }
+
+    private _onSelectedPodInvoked = (event: React.SyntheticEvent<HTMLElement>, pod: V1Pod) => {
+        this.setState({
+            showSelectedPod: true,
+            selectedPod: pod
+        });
+    }
+
+    private static _renderImageCell(rowIndex: number, columnIndex: number, tableColumn: ITableColumn<any>, tableItem: any): JSX.Element {
```
Comment (madhuv-msft) [inline src/WebUI/WorkLoads/WorkloadDetails.tsx]: as per spec this is link

**Comment by madhuv-msft** [comment]
would you want to add UTs also?

File: src/WebUI/Pods/PodsLeftPanel.tsx (hunk 7 above)
Comment (dabasnidhi) [reply src/WebUI/Pods/PodsLeftPanel.tsx]: For all headings across different pages we are using h2, so I am using same here for consistency

Comment (dabasnidhi) [reply src/WebUI/Pods/PodsLeftPanel.tsx]: For all headings across different pages we are using h2, so I am using same here for consistency

File: src/WebUI/Common/KubeSummary.scss (hunk 1 above)
Comment (dabasnidhi) [reply src/WebUI/Common/KubeSummary.scss]: Removed css from id main content, as the height should be set by the parent that is hosting the kubernetes container

File: src/WebUI/Pods/PodOverview.scss (hunk 2 above)
Comment (dabasnidhi) [reply src/WebUI/Pods/PodOverview.scss]: I have moved all platform css to a common file Webplatform.scss

File: src/WebUI/Pods/PodOverview.tsx (hunk 4 above)
Comment (dabasnidhi) [reply src/WebUI/Pods/PodOverview.tsx]: the default css on card is depth-8..But we are using depth-16 everywhere

File: src/WebUI/Pods/PodsLeftPanel.scss (hunk 5 above)
Comment (dabasnidhi) [reply src/WebUI/Pods/PodsLeftPanel.scss]: using fontWeightMedium wherever this was required

File: src/WebUI/Pods/PodsLeftPanel.tsx (hunk 6 above)
Comment (dabasnidhi) [reply src/WebUI/Pods/PodsLeftPanel.tsx]: Added a todo for the same

File: src/WebUI/Pods/PodsRightPanel.tsx (hunk 8 above)
Comment (dabasnidhi) [reply src/WebUI/Pods/PodsRightPanel.tsx]: Added a Todo to move all headers appropriately

File: src/WebUI/Pods/PodsTable.tsx (hunk 9 above)
Comment (dabasnidhi) [reply src/WebUI/Pods/PodsTable.tsx]: yes

File: src/WebUI/WorkLoads/WorkloadDetails.scss (hunk 11 above)
Comment (dabasnidhi) [reply src/WebUI/WorkLoads/WorkloadDetails.scss]: Changed the classnames to  bemore specific

File: src/WebUI/WorkLoads/WorkloadDetails.tsx (hunk 12 above)
Comment (dabasnidhi) [reply src/WebUI/WorkLoads/WorkloadDetails.tsx]: done

File: src/WebUI/WorkLoads/WorkloadDetails.tsx (hunk 15 above)
Comment (dabasnidhi) [reply src/WebUI/WorkLoads/WorkloadDetails.tsx]: Modified Image text according to number of containers, And added to do to make it a link when we have image details view

File: src/WebUI/Pods/PodsTable.tsx (hunk 9 above)
Comment (dabasnidhi) [reply src/WebUI/Pods/PodsTable.tsx]: yes

File: src/WebUI/WorkLoads/WorkloadDetails.tsx (hunk 16 above)
Comment (dabasnidhi) [reply src/WebUI/WorkLoads/WorkloadDetails.tsx]: added to do to make it a link when we have image details view

**Pull request merged**